libs/README.md

```diff
@@ -0,0 +1,375 @@
+
+// api/create_workspace.rs
+fn create_new_workspace(app: AppHandle<Wry>, root_path: Option<String>) -> Result<(), String>
+
+// api/events.rs
+async fn events(app: AppHandle<Wry>) -> Result<(), ()>
+
+// api/host.rs
+async fn check_docker() -> Result<Version, String>
+async fn open_terminal(_app: AppHandle<Wry>, platform: String) -> Result<(), String>
+async fn check_internet_connection() -> Result<bool, String>
+
+// api/launch_docker.rs
+async fn launch_docker(app: AppHandle<Wry>, name: String, config: WorkspaceLaunchOptions) -> Result<(), String>
+
+// api/pull_images.rs
+async fn pull_images(app: AppHandle<Wry>) -> Result<(), String>
+async fn pull_image(app: AppHandle<Wry>, image_name: &str) -> Result<(), String>
+
+// api/service.rs
+async fn start_service(
+    app: AppHandle<Wry>,
+    service_name: String,
+    settings: ServiceSettings,
+) -> Result<StartServiceResult, String>
+async fn stop_service(state: State<'_, AppState>, service_name: String) -> Result<(), String>
+async fn create_default_workspace(app: AppHandle<Wry>, settings: ServiceSettings) -> Result<bool, String>
+
+// api/shutdown.rs
+async fn shutdown(state: State<'_, AppState>) -> Result<String, ()>
+```
+
+All the methods above, in most cases, are adapted methods of Docker.
+For example, the `pull_image` method pulls an image with a specific name,
+and `start_service` starts a particular container. To get events from containers,
+we use the `events` method.
+
+A vast number of errors appear because the frontend part has not considered
+all the exceptional cases: for example, when grpc is not started yet, but
+the frontend tried to connect to it. Or if the **Tor** container is not started,
+but the frontend tried to launch the **Base Node**, that failed with an error
+because it couldn't connect to the **Tor** server.
+
+For the last problem, we have a naive solution that is not solved
+the problem entirely: we added a `WAIT_FOR_TOR` environment variable
+in the **Base Node**, but I often see cases when the tor runs longer
+than that interval expected.
+
+The obvious solution to this problem is: let's just take into account
+all these cases in the frontend part. But how to test all that? To cover
+that cases with tests, we have to create a lot of fragile tests that will
+break simply because we move the buttons to other places.
+
+Another potential problem is the destruction of the main window since
+the frontend part controls the backend. If the main window is destroyed
+accidentally by the system, we lose the state, and the containers
+become unmanaged.
+
+It is also impossible to create a multi-window interface, or
+a container that manages others (a kind of launchpad for servers).
+
+Another class of potential problems is related to the fact that we,
+for example, send SQL queries directly from the frontend to the backend.
+And, in theory, we have to cover that requests with tests as well.
```

Comment:
Note that The SQL queries are for persisting state for the front-end and are not strictly related to the management of containers.